snuba/admin/views.py

```diff
@@ -116,6 +121,52 @@ def clickhouse_system_query() -> Response:
     return make_response(jsonify({"error": "Something went wrong"}), 400)
 
 
+# Sample cURL command:
+#
+# curl -X POST \
+#  -H 'Content-Type: application/json' \
+#  http://localhost:1219/clickhouse_trace_query?query=SELECT+count()+FROM+errors_local
+@application.route("/clickhouse_trace_query", methods=["POST"])
+def clickhouse_trace_query() -> Response:
+    req = json.loads(request.data)
+    try:
+        host = req["host"]
+        port = req["port"]
+        storage = req["storage"]
+        raw_sql = req["sql"]
+    except KeyError as e:
+        return make_response(
+            jsonify({"error": f"Invalid request, missing key {e.args[0]}"}), 400
+        )
+
+    try:
+        result = run_query_and_get_trace(host, port, storage, raw_sql)
+        trace_output = result.trace_output
+        return make_response(jsonify({"trace_output": trace_output}), 200)
+    except InvalidCustomQuery as err:
+        return make_response(jsonify({"error": err.message or "Invalid query"}), 400)
+    except QueryException as err:
+        status = 500
+        details: Mapping[str, Any]
+
+        cause = err.__cause__
```

Comment:
Are you sure this how QueryException is meant to work? It doesn't seem right to get the message out of this private attribute